src/tests/conftest.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+from polydatum import Service
+from polydatum.dal import PathSegment, DataManager
+
+
+@pytest.fixture
+def path_arg():
+    """
+    A factory fixture for a basic PathSegment for use in testing.
+
+    Normally instantiating this for the first time in code implementation
+    will be done by the DAL itself. Because it's somewhat cumbersome, this
+    fixture helps prevent typos and such.
+    """
+    def _factory(name='example'):
+        return (PathSegment(name=name),)
+    return _factory
+
+
+class UserService(Service):
+    def user_test_method(self):
+        return 'user-service'
+
+
+class UserProfileService(Service):
+    def profile_test_method(self):
+        return 'profile-nested-service'
+
+
+class SampleService(Service):
+    def sample_test_method(self):
+        return 'sample-nested-service'
+
+
+class ExampleService(Service):
+    def example_test_method(self):
+        return 'example-nested-service'
+
+
+class DemoService(Service):
+    def demo_test_method(self):
+        return 'demo-nested-service'
+
+
+@pytest.fixture()
+def sub_service_setup():
+    """
+    Sample nested service structure
+    """
+    def _get_dam():
+        data_manager = DataManager()
+        data_manager.register_services(
+            users=UserService().register_services(
+                profile=UserProfileService().register_services(
```

Comment:
Neat. Forgot about this chaining feature. Probably only useful in tests though.

Reply:
Exactly, we actually cribbed this pattern from one of the existing tests. It's repetitive enough that we threw it in a fixture. But yea it's completely useless for anything other than testing the stuff in this PR.